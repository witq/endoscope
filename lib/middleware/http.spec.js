const httpHandler = require("./http.js");

describe("httpHandler", () => {
  const requestMock = {
    method: "GET",
    url: "/healthz"
  };
  const responseMock = {
    writeHead: jest.fn(),
    end: jest.fn()
  };

  beforeEach(() => {
    responseMock.writeHead.mockClear();
    responseMock.end.mockClear();
  });

  it("should respect the successCode option", () => {
    const endoscopeMock = {
      run: jest.fn(() => Promise.resolve([]))
    };

    const handler = httpHandler(endoscopeMock)({ successCode: 666 });

    return handler(requestMock, responseMock).then(() => {
      expect(responseMock.writeHead.mock.calls).toMatchSnapshot();
      expect(responseMock.end.mock.calls).toMatchSnapshot();
    });
  });

  it("should respect the errorCode option", () => {
    const endoscopeMock = {
      run: jest.fn(() => Promise.reject(""))
    };

    const handler = httpHandler(endoscopeMock)({ errorCode: 666 });

    return handler(requestMock, responseMock).then(() => {
      expect(responseMock.writeHead.mock.calls).toMatchSnapshot();
      expect(responseMock.end.mock.calls).toMatchSnapshot();
    });
  });

  it("should return successful result", () => {
    const endoscopeMock = {
      run: jest.fn(() => Promise.resolve(["ok"]))
    };

    const handler = httpHandler(endoscopeMock)();

    return handler(requestMock, responseMock).then(() => {
      expect(responseMock.writeHead.mock.calls).toMatchSnapshot();
      expect(responseMock.end.mock.calls).toMatchSnapshot();
    });
  });

  it("should return error result", () => {
    const endoscopeMock = {
      run: jest.fn(() => Promise.reject("err"))
    };

    const handler = httpHandler(endoscopeMock)();

    return handler(requestMock, responseMock).then(() => {
      expect(responseMock.writeHead.mock.calls).toMatchSnapshot();
      expect(responseMock.end.mock.calls).toMatchSnapshot();
    });
  });

  it("should return 404 status for urls other than prefix", () => {
    const endoscopeMock = {
      run: jest.fn(() => Promise.resolve([]))
    };

    const requestMock = {
      method: "GET",
      url: "/health"
    };

    const handler = httpHandler(endoscopeMock)();

    return handler(requestMock, responseMock).then(() => {
      expect(responseMock.writeHead.mock.calls).toMatchSnapshot();
      expect(responseMock.end.mock.calls).toMatchSnapshot();
    });
  })
});
