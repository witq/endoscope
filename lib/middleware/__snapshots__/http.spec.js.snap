// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`httpHandler should respect the errorCode option 1`] = `
Array [
  Array [
    666,
    Object {
      "Content-Type": "application/json",
    },
  ],
]
`;

exports[`httpHandler should respect the errorCode option 2`] = `
Array [
  Array [
    "",
  ],
]
`;

exports[`httpHandler should respect the successCode option 1`] = `
Array [
  Array [
    666,
    Object {
      "Content-Type": "application/json",
    },
  ],
]
`;

exports[`httpHandler should respect the successCode option 2`] = `
Array [
  Array [
    "[]",
  ],
]
`;

exports[`httpHandler should return 404 status for urls other than prefix 1`] = `
Array [
  Array [
    404,
  ],
]
`;

exports[`httpHandler should return 404 status for urls other than prefix 2`] = `
Array [
  Array [],
]
`;

exports[`httpHandler should return error result 1`] = `
Array [
  Array [
    500,
    Object {
      "Content-Type": "application/json",
    },
  ],
]
`;

exports[`httpHandler should return error result 2`] = `
Array [
  Array [
    "err",
  ],
]
`;

exports[`httpHandler should return successful result 1`] = `
Array [
  Array [
    200,
    Object {
      "Content-Type": "application/json",
    },
  ],
]
`;

exports[`httpHandler should return successful result 2`] = `
Array [
  Array [
    "[\\"ok\\"]",
  ],
]
`;
